tests/beta/test_beta_qnode.py

```diff
@@ -0,0 +1,900 @@
+        assert qn.qtape.measurements == [m1, m2]
+
+    @pytest.mark.xfail
+    def test_multiple_observables_same_wire_expval(self, mocker):
+        """Test that the QNode supports returning expectation values of observables that are on the
+        same wire (provided that they are Pauli words and qubit-wise commuting)"""
+        dev = qml.device("default.qubit", wires=3)
+
+        w = np.random.random((2, 3, 3))
+
+        @qnode(dev)
+        def f(w):
+            qml.templates.StronglyEntanglingLayers(w, wires=range(3))
+            return (
+                qml.expval(qml.PauliX(0)),
+                qml.expval(qml.PauliX(0) @ qml.PauliZ(1)),
+                qml.expval(qml.PauliX(2)),
+            )
+
+        spy = mocker.spy(qml.devices.DefaultQubit, "apply")
+        res = f(w)
+        spy.assert_called_once()
+
+        obs = [qml.PauliX(0), qml.PauliX(0) @ qml.PauliZ(1), qml.PauliX(2)]
+        qnodes = qml.map(qml.templates.StronglyEntanglingLayers, obs, dev)
+        res_2 = qnodes(w)
+
+        assert np.allclose(res, res_2)
+
+    @pytest.mark.xfail
+    def test_multiple_observables_same_wire_mixed(self, mocker):
+        """Test that the QNode supports returning observables that are on the
+        same wire but with different return types (provided that the observables are Pauli words and
+        qubit-wise commuting)"""
+        dev = qml.device("default.qubit", wires=3)
+
+        w = np.random.random((2, 3, 3))
+
+        @qnode(dev)
+        def f(w):
+            qml.templates.StronglyEntanglingLayers(w, wires=range(3))
+            return qml.expval(qml.PauliX(0)), qml.var(qml.PauliX(0) @ qml.PauliZ(1))
+
+        spy = mocker.spy(qml.devices.DefaultQubit, "apply")
+        res = f(w)
+        spy.assert_called_once()
+
+        q1 = qml.map(qml.templates.StronglyEntanglingLayers, [qml.PauliX(0)], dev, measure="expval")
+        q2 = qml.map(
+            qml.templates.StronglyEntanglingLayers,
+            [qml.PauliX(0) @ qml.PauliZ(1)],
+            dev,
+            measure="var",
+        )
+
+        res_2 = np.array([q1(w), q2(w)]).squeeze()
+
+        assert np.allclose(res, res_2)
+
+    def test_operator_all_wires(self, monkeypatch, tol):
+        """Test that a operator that must act on all wires
+        does, or raises an error."""
+        monkeypatch.setattr(qml.RX, "num_wires", qml.operation.AllWires)
+
+        def circuit(x):
+            qml.RX(x, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        dev = qml.device("default.qubit", wires=2)
+        qnode = QNode(circuit, dev)
+
+        with pytest.raises(qml.QuantumFunctionError, match="Operator RX must act on all wires"):
+            qnode(0.5)
+
+        dev = qml.device("default.qubit", wires=1)
+        qnode = QNode(circuit, dev)
+        assert np.allclose(qnode(0.5), np.cos(0.5), atol=tol, rtol=0)
+
+
+class TestDecorator:
+    """Unittests for the decorator"""
+
+    def test_decorator(self, tol):

```

Comment:
This looks almost identical to `test_basic_tape_construction` on 374. Can they be brought together?

Reply:
It does, but I think I would prefer to keep it as a separate test, since it is unit testing a different logical pathway, rather than different test data 🤔 But happy to change it if you think it is worth combining

Reply:
Ok yes I think good testing logic > tidy code. Let's keep as is